### DISCUSSION TITLE
Attempting to introspect the workflow breaks the cache

### DISCUSSION THREAD

**Discussion by dreavjr** [nextflow-io/nextflow#4306] [General]
I'm using nextflow with centralized loggers (e.g., ClearML) and attempting to log the workflow metadata (e.g., ${workflow.runName}) on them. Unfortunaly, it seems that the simple presence of a changeable introspecting variable on a process breaks the cache mechanism, and downstream tasks are no longer resumed.

Is there a pattern to indicate to nextflow that some inputs should not be used to compute the cash hash?

Otherwise, are the workflow metadata stored somewhere accessible from the current path so I could attempt to recover them inside the running process ?

**Comment by bentsherman** [comment]
The `workflow` variable is accessible anywhere in your pipeline script, so you don't have to use it in a process. You could use the `workflow.onComplete` handler to log all of that information once at the end of the workflow.

Or you could have a single process that does it at the beginning. It will be re-run every time because the workflow run name is different every time, but as long as you don't link that process to any downstream processes, the rest of your pipeline should resume normally.

**Comment by dreavjr** [comment]
Thank you for this suggestion! Having some form of "backdoor" access to the `workflow` variables would still be the easiest pattern, but using `workflow.onComplete` might be a second-best solution.
